compare_locales/resource/from_fluent.py

```diff
@@ -0,0 +1,255 @@
+from __future__ import annotations
+
+from collections import OrderedDict
+from typing import Iterator, List, Optional, Tuple, Union
+
+from fluent.syntax import ast as Fluent
+
+from .elements import (
+    CatchallKey,
+    Comment,
+    Entry,
+    Expression,
+    FunctionRef,
+    Junk,
+    Literal,
+    Message,
+    OptionValue,
+    Pattern,
+    PatternMessage,
+    SelectMessage,
+    Text,
+    VariableRef,
+    VariantKey,
+)
+
+
+class SelectArg:
+    def __init__(self, sel: Fluent.SelectExpression) -> None:
+        self.defaultName = ""
+        self.selector = sel.selector
+
+        def getName(v: Fluent.Variant) -> Union[str, None]:
+            name = v.key.name if isinstance(v.key, Fluent.Identifier) else v.key.value
+            if v.default:
+                self.defaultName = str(name)
+                return None
+            else:
+                return name
+
+        self.keys: List[Union[str, int, None]] = list(map(getName, sel.variants))
+
+
+def findSelectArgs(pattern: Fluent.Pattern) -> List[SelectArg]:
+    args: List[SelectArg] = []
+
+    def add(arg: SelectArg) -> None:
+        prev = next((a for a in args if a.selector.equals(arg.selector)), None)
+        if prev:
+            prev.keys += arg.keys
```

Comment:
It isn't lost; thanks to the `.equals()` it needs to be the same for `prev` and `arg`. In the rare case that we've two Fluent selectors within a single message that otherwise match but use a different name for their default variant, we end up considering those selectors as separate in the generated data model so that the `defaultName` isn't lost.

But if that happens, we really ought to complain about it when linting as it's probably a mistake in the source message.